### PR TITLE
Tournament admin notes

### PR DESCRIPTION
**Pull request by hburn7**
Adds tournament admin notes entity

### CODE REVIEW

File: Database/Migrations/20240917023200_Tournament_AdminNotes_Fix.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+        }
+
+        /// <inheritdoc />
+        protected override void Down(MigrationBuilder migrationBuilder)
+        {
+
+        }
+    }
+}
```
Comment (myssto) [inline Database/Migrations/20240917023200_Tournament_AdminNotes_Fix.cs:22]: Blank migration?

File: Database/Entities/AdminNoteEntityBase.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace Database.Entities;
+
+/// <summary>
+/// Base class for an entity that serves as an admin note for an <see cref="IAdminNotableEntity"/>
+/// </summary>
+public abstract class AdminNoteEntityBase : UpdateableEntityBase, IAdminNoteEntity
+{
+    [Column("note")]
+    public string Note { get; set; } = string.Empty;
```
Comment (myssto) [inline Database/Entities/AdminNoteEntityBase.cs:12]: Do we want a size constraint on this?

Comment (hburn7) [reply Database/Entities/AdminNoteEntityBase.cs:12]: Probably not. Mania modding posts incoming lol

File: Database/Entities/TournamentAdminNote.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace Database.Entities;
+
+[Table("tournament_admin_notes")]
+public class TournamentAdminNote : AdminNoteEntityBase
+{
+    public Tournament Tournament { get; } = null!;
```
[the remark this answers is not in the document]
Comment (hburn7) [reply Database/Entities/TournamentAdminNote.cs:8]: Per the docs for `entity.HasMany()`:

```
After calling this method, you should chain a call to WithOne(Expression<Func<TRelatedEntity,TEntity?>>?) to fully configure the relationship. Calling just this method without the chained call will not produce a valid relationship.
```

File: Database/Migrations/20240917023200_Tournament_AdminNotes_Fix.cs (hunk 1 above)
Comment (hburn7) [reply Database/Migrations/20240917023200_Tournament_AdminNotes_Fix.cs]: Changed and tested new migrations.

File: Database/Entities/Tournament.cs [hunk 4]
```diff
@@ -123,4 +124,6 @@ public class Tournament : UpdateableEntityBase, IProcessableEntity, IAuditableEn
 
     [NotMapped]
     public int? ActionBlamedOnUserId { get; set; }
+
+    public ICollection<TournamentAdminNote> AdminNotes { get; set; } = new List<TournamentAdminNote>();
```
[the remark this answers is not in the document]
Comment (hburn7) [reply Database/Entities/Tournament.cs]: Changed

**Review by myssto — APPROVED** [review]

**Pull request merged**